main.tf

```diff
@@ -1,3 +1,95 @@
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_bytes_lite_inputs = [for r0 in var.retention_bytes : (var.plan == "lite" && (r0 < 10485760 || r0 > 104857600) ? tobool("For lite plan the allowed value is [10 MiB, 100 MiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_bytes_standard_inputs = [for r1 in var.retention_bytes : (var.plan == "standard" && (r1 < 102400 || r1 > 1074790400) ? tobool("For standard plan the allowed value is [100 KiB, 1 GiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_bytes_enterprise_inputs = [for r2 in var.retention_bytes : (var.plan == "enterprise-3nodes-2tb" && (r2 < 102400 || r2 > 2199023255552) ? tobool("For enterprise plan the allowed value is [100 KiB, 2 TiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_ms_lite = [for rm0 in var.retention_ms : (var.plan == "lite" && (rm0 < 3600000 || rm0 > 2592000000)) ? tobool("For lite plan allowed range for retention time is [1 hour, 30 days]") : true]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_ms_standard = [for rm1 in var.retention_ms : (var.plan == "standard" && (rm1 < 3600000 || rm1 > 2592000000)) ? tobool("For standard plan allowed range for retention time is [1 hour, 30 days]") : true]
+}
+
+resource "ibm_resource_instance" "es_instance" {
+  name              = var.es_name
+  service           = "messagehub"
+  plan              = var.plan
+  location          = var.region
+  resource_group_id = var.resource_group_id
+  tags              = var.tags
+  timeouts {
+    create = var.create_timeout
+    update = var.update_timeout
+    delete = var.delete_timeout
+  }
+
+  parameters = {
+
+    service-endpoints         = var.service_endpoints
+    private_ip_allowlist      = var.private_ip_allowlist
+    throughput                = var.throughput
+    storage_size              = var.storage_size # Storage capacity cannot be scaled down once instance is created.
+    key_protect_key           = var.key_protect_key_crn
+    backup_encryption_key_crn = local.kp_backup_crn
+  }
+}
+
+##############################################################################
+# SCHEMA
+##############################################################################
+
+resource "ibm_event_streams_schema" "es_schema" {
+  count                = length(var.schemas) * (var.plan == "enterprise-3nodes-2tb" ? 1 : 0)
+  resource_instance_id = ibm_resource_instance.es_instance.id
+  schema_id            = var.schemas[count.index].schema_id
+  schema               = jsonencode(var.schemas[count.index].schema)
+}
+
+##############################################################################
+# TOPIC
+##############################################################################
+
+resource "ibm_event_streams_topic" "es_topic" {
```

Comment:
as per slack discussions, this needs to be reworked so we use 1 topics variable of type list of maps